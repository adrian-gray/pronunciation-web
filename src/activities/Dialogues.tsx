import React from "react";

import SplitHilite from "./../components/SplitHilite";

export default (props: {
  dialogues: string[][];
  ipa: string;
  isUserAuth: boolean;
}) => {
  const { dialogues, ipa } = props;

  const extractLines = (dialogue: string[]) => {
    return dialogue.map((line: string, index: number) => {
      let style = "text-left";
      if (index % 2) style = "text-right";
      const classes = `tight-text list-group-item ${style}`;

      return (
        <li className={classes} key={index}>
          <SplitHilite str={line} />
        </li>
      );
    });
  };

  const expandDialogue = (dialogue: string[], index: number) => (
    <div className="sub-canvas" key={index}>
      <ul className="list-group">{extractLines(dialogue)}</ul>
    </div>
  );

  const lines = dialogues.map(expandDialogue);

  return (
    <div className="headspace">
      <h3>
        {"Short dialogues using "}
        <SplitHilite str={ipa} />
      </h3>
      {lines}
    </div>
  );
};
