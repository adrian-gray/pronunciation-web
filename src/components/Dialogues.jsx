import React from 'react'

import {
  List,
  ListItem,
  ListItemText,
  Paper,
  Typography,
  withStyles
} from '@material-ui/core'

import SplitHilite from './SplitHilite'

const styles = (theme) => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  title: {
    margin: `${theme.spacing.unit * 4}px 0 ${theme.spacing.unit * 2}px`
  }
})

const Dialogues = (props) => {
  const { classes, dialogues, tag } = props
  console.log('classes', classes)
  const extractLines = (dialogue, index) => {
    return dialogue.map((line, index) => (
      <ListItem key={index}>
        <ListItemText>
          <SplitHilite str={line} />
        </ListItemText>
      </ListItem>
    ))
  }

  const expandDialogue = (dialogue, index) => (
    <Paper className={classes.page} key={index}>
      <List className='plain-list'>
        {extractLines(dialogue)}
      </List>
    </Paper>
  )

  const lines = dialogues.map(expandDialogue)

  return (
    <div>
      <Typography variant='title' className={classes.title} gutterBottom>
        {'Short dialogues using '}<SplitHilite str={tag} />
      </Typography>
      {lines}
    </div>
  )
}

export default withStyles(styles)(Dialogues)
