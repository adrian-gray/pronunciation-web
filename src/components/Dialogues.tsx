import React from "react";

import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

import SplitHilite from "./SplitHilite";

const useStyles = makeStyles(theme => ({
  headspace: theme.headspace,
  largeText: theme.largeText,
  subCanvas: theme.subCanvas,
  left: {
    textAlign: "left"
  },
  right: {
    textAlign: "right"
  },
  tight: {
    paddingTop: 0,
    paddingBottom: 0,
    borderBottom: "1px solid #EEE"
  }
}));

const Dialogues = props => {
  const classes = useStyles(props);
  const { dialogues, ipa } = props;

  const extractLines = (dialogue, index) => {
    return dialogue.map((line, index) => {
      let style = classes.left;
      if (index % 2) style = classes.right;

      return (
        <ListItem className={classes.tight} key={index}>
          <ListItemText className={style}>
            <SplitHilite str={line} />
          </ListItemText>
        </ListItem>
      );
    });
  };

  const expandDialogue = (dialogue, index) => (
    <Paper className={classes.subCanvas} key={index}>
      <List>{extractLines(dialogue)}</List>
    </Paper>
  );

  const lines = dialogues.map(expandDialogue);

  return (
    <div className={classes.headspace}>
      <Typography variant="h5" gutterBottom>
        {"Short dialogues using "}
        <SplitHilite str={ipa} />
      </Typography>
      {lines}
    </div>
  );
};

export default withTheme(Dialogues);
