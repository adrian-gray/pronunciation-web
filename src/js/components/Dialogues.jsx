import React from 'react'

import { splitHilite } from './../Utils'

export default function Dialogues (props) {
  function extractLines (dialogue, index) {
    return dialogue.map((line, index) => <li key={index}>{splitHilite(line)}</li>)
  }

  function expandDialogue (dialogue, index) {
    return <ul className='plain-list' key={index}>{extractLines(dialogue)}</ul>
  }

  const lines = props.dialogues.map(expandDialogue)

  return (
    <div className='dialogues'>
      <h3>Short Dialogues</h3>
      {lines}
    </div>
  )
}
