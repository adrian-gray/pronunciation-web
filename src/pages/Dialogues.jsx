import React from 'react'

import {
  withStyles
} from '@material-ui/core'

const styles = (theme) => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
    flexGrow: 1
  }
})

const Dialogues = (props) => {
  return (
    <div>Dialogues</div>
  )
}

export default withStyles(styles)(Dialogues)
