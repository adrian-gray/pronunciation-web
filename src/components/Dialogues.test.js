import React from "react";
import { cleanup, render } from "@testing-library/react";

import Dialogues from "./Dialogues";

afterEach(cleanup);

test("Dialogues", async () => {
  const props = {
    ipa: "a:",
    dialogues: [
      ["foo", "bar", "baz", "quox"],
      ["apple", "banana", "chad", "elle"]
    ],
    isUserAuth: true
  };

  const { container } = render(<Dialogues {...props} />);

  expect(container).toMatchSnapshot();
});
