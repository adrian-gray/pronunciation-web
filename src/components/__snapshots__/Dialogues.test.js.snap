// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialogues 1`] = `
<div>
  <div
    class="makeStyles-headspace-1"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Short dialogues using 
      <span>
        a:
      </span>
       
    </h5>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-subCanvas-3 MuiPaper-rounded"
    >
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-left-4"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                foo
              </span>
               
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-right-5"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                bar
              </span>
               
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-left-4"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                baz
              </span>
               
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-right-5"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                quox
              </span>
               
            </span>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-subCanvas-3 MuiPaper-rounded"
    >
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-left-4"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                apple
              </span>
               
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-right-5"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                banana
              </span>
               
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-left-4"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                chad
              </span>
               
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-tight-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root makeStyles-right-5"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              <span>
                elle
              </span>
               
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
